File: backend/app/schemas/promotion.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class PromotionOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    service_id: uuid.UUID
    name: str
    discount_type: str
    discount_value: int
    starts_at: datetime
    ends_at: datetime
    is_active: bool


class PromotionCreate(BaseModel):
    service_id: uuid.UUID
    name: str = Field(min_length=1, max_length=200)
    discount_type: Literal["percentage", "fixed"]
    discount_value: int = Field(ge=0)
    starts_at: datetime
    ends_at: datetime


class PromotionUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=200)
    discount_type: Literal["percentage", "fixed"] | None = None
    discount_value: int | None = Field(default=None, ge=0)
    starts_at: datetime | None = None
    ends_at: datetime | None = None
    is_active: bool | None = None

File: app_admin/lib/models/promotion.dart
class Promotion {
  const Promotion({
    required this.id,
    required this.serviceId,
    required this.name,
    required this.discountType,
    required this.discountValue,
    required this.startsAt,
    required this.endsAt,
    required this.isActive,
  });

  final String id;
  final String serviceId;
  final String name;

  /// 'percentage' ou 'fixed' — ver backend/app/schemas/promotion.py.
  final String discountType;
  final int discountValue;
  final DateTime startsAt;
  final DateTime endsAt;
  final bool isActive;

  factory Promotion.fromJson(Map<String, dynamic> json) {
    return Promotion(
      id: json['id'] as String,
      serviceId: json['service_id'] as String,
      name: json['name'] as String,
      discountType: json['discount_type'] as String,
      discountValue: json['discount_value'] as int,
      startsAt: DateTime.parse(json['starts_at'] as String),
      endsAt: DateTime.parse(json['ends_at'] as String),
      isActive: json['is_active'] as bool,
    );
  }
}
